### ISSUE TITLE
Project Settings: Create Project Settings Dialog

### ISSUE THREAD

**Issue by tjcouch-sil** [paranext/paranext-core#867]
As a user, I want to be able to adjust project settings that affect checks so I can use checks.

----

- [x] Create a Project Settings dialog
- [x] Must accept project settings object
- [x] Display settings groups and individual settings according to settings JSON
- [x] Display settings using reasonable components based on the data type of the default setting value
  - [x] if the default value is a string, give an input field
  - [x] if the default value is an object, give a text field that is the value stringified with space between the lines (`JSON.stringify(value, undefined, 2)` for example)
  - [x] number
  - [x] boolean
- [x] When someone changes a value, immediately update the setting (maybe do a debounce so it applies a bit after they change it? dunno - maybe ask UX)
- [x] If validation throws, show the error message on the screen next to the component. If validation returns false, show some generic error message

[Settings & State Design Doc](https://docs.google.com/document/d/1dUt4cj1VK1f-xchdynERL-OO9SjDEKU4dVF3KfSRwXc/edit)

**Comment by katherinejensen00** [comment]
https://github.com/sillsdev/config-r may have some helpful insights

**Comment by roopa0222** [comment]
Repos Steps.

I have updated the `paranext-core` with the latest main code.

In the test extension code was implemented to validate the User settings and Project Settings.
1. On the main PB hamburger menu click on the Settings menu item
2. Verify that the User Settings Dialog displays the Settings provided in settings.json from Test extension
3. On the main PB hamburger menu click on the 'Open PAPI Menu TestScreen' menu item
4. Click on the webview menu 'Open Project Settings' menu item
5. Verify that the Project Settings Dialog displays the Project Settings provided in projectSettins.json from Test extension

![UserSettings_Valid](https://github.com/user-attachments/assets/627523b5-f217-4734-b8b7-26acf49d4579)

![UserSetting_Invalid](https://github.com/user-attachments/assets/bc8d3929-14dd-4364-8c42-214a4298df67)

![ProjectSetting_Invalid](https://github.com/user-attachments/assets/0e9b095f-052c-4371-a00e-39be984c873e)
